### PR TITLE
Add ParseField instance for Data.ByteString.Char8

### PR DESCRIPTION
**Pull request by bschwb**
I tried to implement #7.
Please check if this is the desired behavior, as I don't have any experience with `ByteString`.

### CODE REVIEW

**Comment by Gabriella439** [comment]
Could you reuse the instance for `Text` and use `Data.Text.Encoding.encodeUTF8` to transform the `Text` result to a `ByteString`?

Also, could you add an instance for lazy `ByteString`s, too?

I'm still trying to decide if we should wrap this in a newtype or not to define which encoding to use.  In other words, something like:

``` haskell
newtype UTF8 a = UTF8 a

instance ParseField (UTF8 ByteString) where
    ...
```

**Comment by bschwb** [comment]
Thank you for the feedback!

Which other encodings do you want to support?

File: src/Options/Generic.hs [hunk 1]
```diff
@@ -293,11 +297,23 @@ parseString metavar m =
...
 
+parseText :: String -> Maybe Text -> Parser Data.Text.Text
+parseText metavar = fmap (fmap Data.Text.pack) (parseString metavar)
+
 instance ParseField Data.Text.Text where
-    parseField = fmap (fmap Data.Text.pack) (parseString "TEXT")
+    parseField = parseText "TEXT"
+
+instance ParseField Data.ByteString.ByteString where
+    parseField = fmap (fmap (Data.Text.Encoding.encodeUtf8)) (parseText "UTF8-BYTESTRING")
```
Comment (Gabriella439) [inline src/Options/Generic.hs]: I would still have the field meta-var be "TEXT".  From the command-line user's perspective there is no difference in what they enter.  Only the programmer decoding the result cares that it is encoded into a `ByteString` in a UTF8 format.

**Comment by Gabriella439** [comment]
I think just handling the UTF8 encoding is good enough for now.  If people would like other encodings they can submit pull requests to add them later on

**Comment by Gabriella439** [comment]
Could you also add a `ParseFields` instance for each type?  The default implementation should do the right thing, so you should only need to write:

``` haskell
instance ParseFields Data.ByteString.ByteString
instance ParseFields Data.ByteString.Lazy.ByteString
```

**Comment by Gabriella439** [comment]
Whoops!  I missed that you already added those instances.  I will go ahead and resolve the merge conflict and merge this in

**Pull request merged**